### PR TITLE
Fix various issues with PredisCache and RedisCache

### PR DESCRIPTION
**Pull request by zeroedin-bill**
Sorry for PR spam :laughing:

### CODE REVIEW

File: lib/Doctrine/Common/Cache/PredisCache.php [hunk 1]
```diff
@@ -61,6 +61,7 @@ protected function doContains($id)
      */
     protected function doSave($id, $data, $lifeTime = 0)
     {
+        $data = serialize($data);
```
Comment (Ocramius) [inline lib/Doctrine/Common/Cache/PredisCache.php]: Empty line after zis

**Comment by Ocramius** [comment]
:+1:

File: tests/Doctrine/Tests/Common/Cache/CacheTest.php [hunk 2]
```diff
@@ -307,6 +309,52 @@ public function falseCastedValuesProvider()
...
+        $cache->save("obj1", $obj1);
+        $obj2 = new \stdClass();
+        $obj2->bar = "baz";
+        $cache->save("obj2", $obj2);
+
+        $fetched = $cache->fetchMultiple(array("obj1", "obj2"));
+        $this->assertInstanceOf("stdClass", $fetched["obj1"]);
+        $this->assertInstanceOf("stdClass", $fetched["obj2"]);
+        $this->assertEquals("bar", $fetched["obj1"]->foo);
+        $this->assertEquals("baz", $fetched["obj2"]->bar);
```
Comment (vvondra) [inline tests/Doctrine/Tests/Common/Cache/CacheTest.php]: I'd test cache misses as well

**Comment by razvanphp** [comment]
Can you please tag this into a new minor version? It's been up there for more than a month.
We are using the master because of this (serious) bug.
Thank you!

**Comment by davedevelopment** [comment]
I just got stung by this, could probably have done with something in the changelog? A hot cache that had already serialized data in was returning incorrect objects etc, when application code was expecting serialized data. 

Anyway, not to worry, but if anyone else comes confused about problems going from 1.4.1 to 1.4.2, it could be this.

**Pull request merged**